[FIX] generate png for restart view

FazleRabbbiferdaus172 opened the pull request:
## Description

**_run_screen_** need to be executed by screenshot generator test to generate _png_ for restart view. But we don't want to restart in case of **_ScreenshotRenderer_**. So, only running the **_DoResetThread_** when **_is_screenshot_renderer_** is false should be enough. The png file is saved in the intended destination in the _show_image_ method of _ScreenshotRenderer_ class which get executed by _run_screen_. Before this pr the the _run_ method of **_RestartView_** was being returned before **_run_screen_** method could be executed as a result the png was not being generated. 

This pull request is categorized as a:

- [ ] New feature
- [x] Bug fix
- [ ] Code refactor
- [ ] Documentation
- [ ] Other

## Checklist

- [x] I’ve run `pytest` and made sure all unit tests pass before sumbitting the PR

If you modified or added functionality/workflow, did you add new unit tests?

- [ ] No, I’m a fool
- [ ] Yes
- [x] N/A

I have tested this PR on the following platforms/os:

- [ ] Raspberry Pi OS [Manual Build](https://github.com/SeedSigner/seedsigner/blob/dev/docs/manual_installation.md)
- [ ] [SeedSigner OS](https://github.com/SeedSigner/seedsigner-os) on a Pi0/Pi0W board
- [x] Other

FazleRabbbiferdaus172 committed:
[FIX] generate png for restart view

run_screen need to be executed by screenshot generator test to generate png for restart view. But we don't want to restart so the not starting the thread should be the way.

Chaitanya-Keyal commented:
Thanks for this PR! This bug seems to have been overlooked in the changes from #758. It went unnoticed because the screenshot generator didn’t raise any errors, it just failed silently without creating a new screenshot. It might be worth debugging that behavior as well.

Chaitanya-Keyal commented on src/seedsigner/views/view.py:
```diff
@@ -255,15 +255,14 @@ def __post_init__(self):
         from seedsigner.gui.screens.screen import ResetScreen
 
-        self.thread.start()
+        if not self.is_screenshot_renderer:
+            # For the screenshot generator, we don't actually want to restart
+            self.thread.start()
```
```suggestion
        if not self.is_screenshot_renderer:
            # For the screenshot generator, we don't actually want to restart
            logger.info("Restarting SeedSigner")
            self.thread.start()
```

Moving the log statement inside the `if` block is clearer, since otherwise the screenshot generator would log a restart even though no restart actually happens. [See this review comment](https://github.com/SeedSigner/seedsigner/pull/758#discussion_r2234075758)

FazleRabbbiferdaus172 replied:
Thanks for the suggestion, I was not entirely sure if where to put the log so did not change but I would prefer if the log was moved to **DoResetThread** just before the sleep. Which keeps everything in place also it indicates the thread responsible for restarting is being executed.

FazleRabbbiferdaus172 commented:
> Thanks for this PR! This bug seems to have been overlooked in the changes from #758. It went unnoticed because the screenshot generator didn’t raise any errors, it just failed silently without creating a new screenshot. It might be worth debugging that behavior as well.

currently there is no check for that. Generator expect the image file will always be generated. It can easily be checked when **ScreenshotComplete** is caught by the generator.

FazleRabbbiferdaus172 committed:
log moved to DoResetThread

Placing the log message inside the thread means it gets written at the exact moment the restart process actually begins. As in case of restart view this never gets executed. Better encapsulates the component responsible for rest.

FazleRabbbiferdaus172 committed:
explicit file regeneration check to screenshot generator

Previously, there were no checks to determine if a file was correctly generated by the generator. It was even hard to detect because of previously generated files.

FazleRabbbiferdaus172 commented:
Added the check for file regeneration

Chaitanya-Keyal commented on tests/screenshot_generator/generator.py:
```diff
@@ -479,13 +479,26 @@ def screencap_view(screenshot_config: ScreenshotConfig):
             screencap_view(screenshot_config)
+
+            if not os.path.exists(screenshot_filepath):
+                raise Exception(f"Failed to generate screenshot for '{screenshot_config.screenshot_name}' in section '{section_name}'.")
```
```suggestion
                raise Exception(f"No screen was rendered for '{screenshot_config.screenshot_name}' in section '{section_name}'.")
```

Nit: Perhaps change this error message to say 'No screen was rendered' (or something similar), instead of 'Failed to generate', since the latter makes it sound like an issue with the screenshot generator code, when the problem is actually with the view itself.

Chaitanya-Keyal reviewed (approved):
Tested ACK.

kdmukai commented on src/seedsigner/views/view.py:
```diff
@@ -275,6 +272,7 @@ def run(self):
 
             # Give the screen just enough time to display the reset message before
             # exiting.
+            logger.info("Restarting SeedSigner")
```
This is splitting the comment from the line of code it's explaining. I would move it above the comment.

kdmukai commented:
I wasn't paying close enough attention to the light refactor in `RestartView` in #758. The move to push the thread instantiation into the `__post_init__()` just isn't necessary. I'd rather see us drop that change and just collect all the fixes in `run()`:

```python
    def run(self):
        from seedsigner.gui.screens.screen import ResetScreen

        if not self.is_screenshot_renderer:
            # For the screenshot generator, we don't actually want to restart
            RestartView.DoResetThread().start()

        self.run_screen(ResetScreen)
```

kdmukai commented on src/seedsigner/views/view.py:
```diff
@@ -255,15 +255,12 @@ def __post_init__(self):
 
-        self.thread.start()
+        if not self.is_screenshot_renderer:
+            # For the screenshot generator, we don't actually want to restart
```
This phrasing is awkward. It's always difficult with a `not` condition. Maybe something like:

`# We don't want the screenshot generator to actually try to do the restart`

kdmukai commented:
I think I'd prefer to keep this PR dead simple and avoid the changes in `generator.py` to detect when a screenshot fails to render.

I have another approach in mind to alert us when a screenshot isn't rendered. I'll create a PR for it and link to it back here.

FazleRabbbiferdaus172 commented:
> I think I'd prefer to keep this PR dead simple and avoid the changes in `generator.py` to detect when a screenshot fails to render.
> 
> I have another approach in mind to alert us when a screenshot isn't rendered. I'll create a PR for it and link to it back here.

I will revert the generator.py

FazleRabbbiferdaus172 committed:
Revert "explicit file regeneration check to screenshot generator"

This reverts commit e32604c1a60668079b207c0f00c9879ae5dfb651.

FazleRabbbiferdaus172 committed:
[REF] __post_init__ removed and RestartView initiates and start inside run method

FazleRabbbiferdaus172 committed:
[REF] logger in is moved above the comment

FazleRabbbiferdaus172 commented:
suggested changes are pushed
